esp-radio/src/wifi/mod.rs

```diff
@@ -2494,6 +2516,16 @@ impl WifiController<'_> {
         }
     }
 
+    /// Wait for the station to disconnect.
+    pub async fn wait_for_station_disconnect(&mut self) {
```

Comment:
I guess the idea is in the proper impl this will have state checks to ensure we're connected or in a state that emit the disconnection event?